rascal2/widgets/project.py

```diff
@@ -0,0 +1,278 @@
+        self.save_project_button.setIcon(QtGui.QIcon(path_for("save-project.png")))
+        self.save_project_button.clicked.connect(self.save_changes)
+
+        self.cancel_button = QtWidgets.QPushButton(" Cancel", self, objectName="redbutton")
+        self.cancel_button.setIcon(QtGui.QIcon(path_for("cancel-dark.png")))
+        self.cancel_button.clicked.connect(self.cancel_changes)
+
+        layout = QtWidgets.QHBoxLayout()
+        layout.setAlignment(QtCore.Qt.AlignmentFlag.AlignRight)
+        layout.addWidget(self.save_project_button)
+        layout.addWidget(self.cancel_button)
+        main_layout.addLayout(layout)
+
+        self.edit_calculation_label = QtWidgets.QLabel("Calculation:", self, objectName="boldlabel")
+
+        self.calculation_combobox = QtWidgets.QComboBox(self)
+        self.calculation_combobox.addItems([calc for calc in Calculations])
+
+        layout = QtWidgets.QHBoxLayout()
+        layout.addWidget(self.edit_calculation_label)
+        layout.addWidget(self.calculation_combobox)
+
+        self.edit_model_type_label = QtWidgets.QLabel("Model Type:", self, objectName="boldlabel")
+
+        self.model_combobox = QtWidgets.QComboBox(self)
+        self.model_combobox.addItems([model for model in LayerModels])
+
+        layout.addWidget(self.edit_model_type_label)
+        layout.addWidget(self.model_combobox)
+
+        self.edit_geometry_label = QtWidgets.QLabel("Geometry:", self, objectName="boldlabel")
+
+        self.geometry_combobox = QtWidgets.QComboBox(self)
+        self.geometry_combobox.addItems([geo for geo in Geometries])
+
+        self.calculation_combobox.currentTextChanged.connect(
+            partial(self.on_combobox_update, self.calculation_combobox, "inter_calculation")
+        )
+        self.model_combobox.currentTextChanged.connect(
+            partial(self.on_combobox_update, self.model_combobox, "inter_model")
+        )
+        self.geometry_combobox.currentTextChanged.connect(
+            partial(self.on_combobox_update, self.geometry_combobox, "inter_geometry")
+        )
+
+        layout.addWidget(self.edit_geometry_label)
+        layout.addWidget(self.geometry_combobox)
+        main_layout.addLayout(layout)
+
+        self.edit_project_tab = QtWidgets.QTabWidget()
+
+        # Replace QtWidgets.QWidget() with methods to create
+        # the tabs in edit view.
+        self.edit_project_tab.addTab(QtWidgets.QWidget(), "Parameters")
+        self.edit_project_tab.addTab(QtWidgets.QWidget(), "Experimental Parameters")
+        self.edit_project_tab.addTab(QtWidgets.QWidget(), "Layers")
+        self.edit_project_tab.addTab(QtWidgets.QWidget(), "Data")
+        self.edit_project_tab.addTab(QtWidgets.QWidget(), "Contrasts")
+
+        main_layout.addWidget(self.edit_project_tab)
+
+        self.edit_project_widget.setLayout(main_layout)
+
+    def on_combobox_update(self, combo_box: QtWidgets.QComboBox, attr_name: str, selected_value: str) -> None:
```

Comment:
function names should describe an action, e.g. `process_combobox_update` or something